File: benchmark/python/matrix.py
import time

size = 150
a = []
b = []
c = []
for i in range(size):
    row_a = []
    row_b = []
    row_c = []
    for j in range(size):
        row_a.append(i + j)
        row_b.append(i - j)
        row_c.append(0)
    a.append(row_a)
    b.append(row_b)
    c.append(row_c)

start = time.time() * 1000
for i in range(size):
    for j in range(size):
        for k in range(size):
            c[i][j] = c[i][j] + a[i][k] * b[k][j]
end = time.time() * 1000
print(f"Result[0][0]: {c[0][0]}")
print(f"Result[149][149]: {c[149][149]}")
print(f"Time: {end - start:.0f}ms")

File: benchmark/php/matrix.php
<?php
$size = 150;
$a = []; $b = []; $c = [];
for ($i = 0; $i < $size; $i++) {
    $ra = []; $rb = []; $rc = [];
    for ($j = 0; $j < $size; $j++) {
        $ra[] = $i + $j; $rb[] = $i - $j; $rc[] = 0;
    }
    $a[] = $ra; $b[] = $rb; $c[] = $rc;
}
$start = hrtime(true);
for ($i = 0; $i < $size; $i++) {
    for ($j = 0; $j < $size; $j++) {
        $s = 0;
        for ($k = 0; $k < $size; $k++) { $s += $a[$i][$k] * $b[$k][$j]; }
        $c[$i][$j] = $s;
    }
}
$end = hrtime(true);
$ms = ($end - $start) / 1e6;
echo "Result[0][0]: " . $c[0][0] . "
";
echo "Result[149][149]: " . $c[149][149] . "
";
echo "Time: " . round($ms) . "ms
";
